Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,317 @@
+    /// </summary>
+    /// <returns>A string representing a random request ID in the format used by API Gateway for HTTP APIs.</returns>
+    /// <remarks>
+    /// The generated ID is a 145character string consisting of lowercase letters and numbers, followed by an equals sign.
+    private static string GenerateRequestId()
+    {
+        return Guid.NewGuid().ToString("N").Substring(0, 8) + Guid.NewGuid().ToString("N").Substring(0, 7) + "=";
+    }
+
+    /// <summary>
+    /// Sets the response body on the HttpResponse.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the body on.</param>
+    /// <param name="body">The body content.</param>
+    /// <param name="isBase64Encoded">Whether the body is Base64 encoded.</param>
+    private static void SetResponseBody(HttpResponse response, string? body, bool isBase64Encoded, ApiGatewayEmulatorMode apiGatewayEmulator)
+    {
+        if (!string.IsNullOrEmpty(body))
+        {
+            byte[] bodyBytes;
+            if (isBase64Encoded && ApiGatewayEmulatorMode.Rest != apiGatewayEmulator) // rest api gateway doesnt automatically decode the response
+            {
+                bodyBytes = Convert.FromBase64String(body);
+            }
+            else
+            {
+                bodyBytes = Encoding.UTF8.GetBytes(body);
+            }
+
+            response.Body = new MemoryStream(bodyBytes);
+            response.ContentLength = bodyBytes.Length;
+        }
+    }
+
+    /// <summary>
+    /// Sets the content type and status code for API Gateway v1 responses.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the content type and status code on.</param>
+    /// <param name="headers">The single-value headers.</param>
+    /// <param name="multiValueHeaders">The multi-value headers.</param>
+    /// <param name="statusCode">The status code to set.</param>
+    private static void SetContentTypeAndStatusCode(HttpResponse response, IDictionary<string, string>? headers, IDictionary<string, IList<string>>? multiValueHeaders, int statusCode, ApiGatewayEmulatorMode emulatorMode)
+    {
+        string? contentType = null;
+
+        if (headers != null && headers.TryGetValue("Content-Type", out var headerContentType))
+        {
+            contentType = headerContentType;
+        }
+        else if (multiValueHeaders != null && multiValueHeaders.TryGetValue("Content-Type", out var multiValueContentType))
+        {
+            contentType = multiValueContentType[0];
+        }
+
+        if (contentType != null)
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.HttpV1)
+            {
+                response.ContentType = "text/plain; charset=utf-8";
+            } 
+            else if (emulatorMode == ApiGatewayEmulatorMode.Rest)
+            {
+                response.ContentType = "application/json";
+            }
```

Comment:
for the sake of completion, add an else and handle the fact that we could mistakenly make this call for V2. Maybe throw an exception as this is dev error and needs to be updated.

Reply:
fixed